File: Núcleo/src/LogicaNegocio/Medico.java
package LogicaNegocio;

import Recursos.DatosMedico;
import Recursos.DatosPaciente;
import Recursos.DatosPersonal;
import java.util.Date;

/**
 * @author Victor Javier
 * @version 1.0
 * @created 18-may.-2019 06:27:02 p. m.
 */
public class Medico extends Personal {

    IMedico iMedico;
    
    public Medico() {

    }

    public Medico(IMedico iMedico) {
        this.iMedico = iMedico;
    }

    public IMedico getiMedico() {
        return iMedico;
    }

    public void setiMedico(IMedico iMedico) {
        this.iMedico = iMedico;
    }
    private DatosPersonal validarDatos(){
        DatosPersonal datosMedico = DatosPersonal.VALIDO;
        if (this.nombre == null || this.nombre.trim().equals("")){
            datosMedico = DatosPersonal.NOMBRE_VACIO;
        } else if (this.nombre.trim().length() > 70){
            datosMedico = DatosPersonal.NOMBRE_LARGO;
        } else if (this.apellido.trim().equals("") || this.apellido == null){
            datosMedico = DatosPersonal.APELLIDO_VACIO;
        } else if (this.apellido.trim().length() > 70){
            datosMedico = DatosPersonal.APELLIDO_LARGO;
        } else if (this.rfc.trim().equals("")){
            datosMedico = DatosPersonal.RFC_VACIO;
        } else if (this.fechaNacimiento == null){
            datosMedico = DatosPersonal.FECHA_NACIMIENTO_VACIA;
        } else if (this.sexo == ' '){
            datosMedico = DatosPersonal.SEXO_VACIO;
        } else if (this.numeroTelefono.trim().equals("") || this.numeroTelefono == null){
            datosMedico = DatosPersonal.NUMERO_TELEFONO_VACIO;
        } else if (this.numeroTelefono.trim().length() > 12){
            datosMedico = DatosPersonal.NUMERO_TELEFONO_LARGO;
        } else if(this.numeroTelefono.trim().length() < 10){
            datosMedico = DatosPersonal.NUMERO_TELEFONO_CORTO;
        } else if (this.numeroPersonal.trim().equals("") || this.numeroPersonal == null){
            datosMedico = DatosPersonal.NUMERO_PERSONAL_VACIO;
        } else if(this.turno.trim().equals("") || this.turno == null){
            datosMedico = DatosPersonal.TURNO_VACIO;
        }
        return datosMedico;
    }
    
    /*public DatosPaciente registrarMedico() {
        DatosPaciente registro = this.validarDatos();
        if (registro == DatosPaciente.VALIDO){
            if (this.obtenerMedico(this.numeroPersonal) == null){
                registro = this.iMedico.registrarMedico(this) ? DatosPaciente.EXITO : DatosPaciente.ERROR_ALMACENAMIENTO;
            }
        }
        return registro;
    }*/

    /*public DatosPaciente modificarMedico() {
        DatosPaciente registro = this.validarDatos();
        if (registro == DatosPaciente.VALIDO){
                registro = this.iMedico.modificarMedico(this) ? DatosPaciente.EXITO : DatosPaciente.ERROR_ALMACENAMIENTO;
            
        }
        return registro;
    }*/

    public Medico obtenerMedico(String numeroPersonal) {
        return iMedico.obtenerMedico(numeroPersonal);
    }

    /**
     *
     * @param numeroConsultorio
     * @return 
     */
    @Override
    public boolean registrarEntrada(String numeroConsultorio) {
        boolean respuesta = false;
        if (!numeroConsultorio.trim().equals("") || numeroConsultorio != null ) {
            respuesta = iMedico.registrarEntrada(numeroConsultorio, this.numeroPersonal );
        }
        return respuesta;
    }

    /**
     *
     * @return 
     */
    @Override
    public boolean registrarSalida() {
        return iMedico.registrarSalida(numeroPersonal);
    }

    @Override
    public DatosPersonal registrar() {
        DatosPersonal registro = this.validarDatos();
        if (registro == DatosPersonal.VALIDO){
            if (this.obtenerMedico(this.numeroPersonal) == null){
                registro = this.iMedico.registrar(this) ? DatosPersonal.EXITO : DatosPersonal.ERROR_ALMACENAMIENTO;
            }
        }
        return registro;
    }

    @Override
    public DatosPersonal modificar() {
        DatosPersonal registro = this.validarDatos();
        if (registro == DatosPersonal.VALIDO){
                registro = this.iMedico.modificar(this) ? DatosPersonal.EXITO : DatosPersonal.ERROR_ALMACENAMIENTO;
            
        }
        return registro;
    }

    @Override
    public boolean eliminar() {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    @Override
    public Personal obtenerPersonal(String numeroPersonal) {
        return iMedico.obtenerMedico(numeroPersonal);
    }
}//end Medico

File: Nucleo servidor web/application/libraries/Medico.php
<?php

if ( ! defined('BASEPATH')) exit('No direct script access allowed');
require_once(APPPATH.'libraries/recursos/DatosPersonal.php');
require_once(APPPATH.'libraries/Personal.php');

class Medico extends Personal{
    private $iMedico;

    public function __get($attr) {
		return CI_Controller::get_instance()->$attr;
    }
    
    public function __construct(){
        
    }

    private function validarDatosPersonalesMedico(){
        $datosMedico = DatosPersonal::VALIDO;
        if($this->getNumeroTelefono() === "" || $this->getNumeroTelefono() === null){
            $datosMedico = DatosPersonal::NUMERO_TELEFONO_VACIO;
        }else if(strlen($this->getNumeroTelefono()) > 10){
            $datosMedico = DatosPersonal::NUMERO_TELEFONO_LARGO;
        }else if(strlen($this->getNumeroTelefono()) < 10){
            $datosMedico = DatosPersonal::UMERO_TELEFONO_CORTO;
        }else if($this->getNombre() === "" || $this->getNombre() === null){
            $datosMedico = DatosPersonal::NOMBRE_VACIO;
        }else if(strlen($this->getNombre()) > 70){
            $datosMedico = DatosPersonal::NOMBRE_LARGO;
        }else if($this->getApellido()==="" || $this->getApellido() === null){
            $datosMedico = DatosPersonal::APELLIDO_VACIO;
        }else if(strlen($this->getApellido()) > 70){
            $datosMedico = DatosPersonal::APELLIDO_LARGO;
        }else if($this->getFechaNacimiento() === "" || $this->getFechaNacimiento() === null){
            $datosMedico = DatosPersonal::FECHA_VACIA;
        }else if($this->getSexo()==""){
            $datosMedico = DatosPersonal::SEXO_VACIO;
        }else  if($this->getRfc()==="" || $this->getRfc() === null){
            $datosMedico = DatosPersonal::RFC_VACIO;
        }else if(strlen($this->getRfc()) > 13){
            $datosMedico = DatosPersonal::RFC_LARGO;
        }else if(strlen($this->getRfc())< 13){
            $datosMedico = DatosPersonal::RFC_CORTO;
        }else if($this->getNumeroPersonal()==="" || $this->getNumeroPersonal() === null){
            $datosMedico = DatosPersonal::NUMERO_PERSONAL_VACIO;
        }else if($this->getTurno()==="" || $this->getTurno() == null){
            $datosMedico = DatosPersonal::TURNO_VACIO;
        }
        return $datosMedico;
    }

    public function setIMedico($iMedico){
        $this->iMedico = $iMedico;
    }

    public function registrar($idUsuario){
        $registro = $this->validarDatosPersonalesMedico();
        if($registro === DatosPersonal::VALIDO){
            if($this->obtenerPersonal($this->getNumeroPersonal())== null){
                if($this->iMedico->registrar($this, $idUsuario)){
                    $registro = DatosPersonal::EXITO;
                }else{
                    $registro = DatosPersonal::ERROR_ALMACENAMIENTO;
                }
            }
            return $registro;
        }
    }

    public function modificar(){
        $registro = $this->validarDatos();
        if($registro === DatosPersonal::VALIDO){
            if($this->iMedico->modificar($this)){
                $registro = DatosPersonal::EXITO;
            }else{
                $registro = DatosPersonal::ERROR_ALMACENAMIENTO;
            }
        }
        return $registro;
    }

    public function registrarEntrada($numeroConsultorio){
        $respuesta = false;
        if($numeroConsultorio!="" | $numeroConsultorio!= null){
            $respuesta = $this->iMedico->registrarEntrada($this->getRfc(), $numeroConsultorio);
        }
        return $respuesta;
    }

    public function registrarSalida(){
        return $this->iMedico->registrarSalida($this->getRfc());
    }

    public function eliminar(){
        return $this->iMedico->eliminar($this->getRfc());
    }

    public function obtenerPersonal($rfc){
        return $this->iMedico->obtenerMedico($rfc);
    }
    public function obtenerPersonalId($idUsuario){
        return $this->iMedico->obtenerMedicoId($idUsuario);
    }

}